### PR TITLE
Add logic operators to MemoryArbitrator::Stats

### PR DESCRIPTION
**Pull request by tanjialiang**
The logic operators will be used for metrics maintaining and exporting in Presto native.

### CODE REVIEW

**Review by xiaoxmeng — APPROVED** [review]
@tanjialiang LGTM. Thanks!

File: velox/common/memory/MemoryArbitrator.h [hunk 1]
```diff
@@ -189,6 +211,12 @@ class MemoryArbitrator {
   const uint64_t memoryPoolTransferCapacity_;
 };
 
+FOLLY_ALWAYS_INLINE std::ostream& operator<<(
+    std::ostream& o,
+    const MemoryArbitrator::Stats& stats) {
+  return o << stats.toString();
```
Comment (xiaoxmeng) [inline velox/common/memory/MemoryArbitrator.h:217]:
```
o << stats.toString();
return o;
```

File: velox/common/memory/MemoryArbitrator.h [hunk 2]
```diff
@@ -170,9 +170,31 @@ class MemoryArbitrator {
...
+        uint64_t _queueTimeUs,
+        uint64_t _arbitrationTimeUs,
+        uint64_t _numShrunkBytes,
+        uint64_t _numReclaimedBytes,
+        uint64_t _maxCapacityBytes,
+        uint64_t _freeCapacityBytes);
+
+    Stats() = default;
+
+    Stats operator-(const Stats& other) const;
```
Comment (xiaoxmeng) [inline velox/common/memory/MemoryArbitrator.h:186]: Can you comment about the special treatment on max capacity and free capacity? Thanks!

**Review by xiaoxmeng — APPROVED** [review]
@tanjialiang LGTM. Thanks for the update!

File: velox/common/memory/MemoryArbitrator.h (hunk 1 above)
Comment (tanjialiang) [reply velox/common/memory/MemoryArbitrator.h:217]: What is the difference here?

**Pull request closed without merging**